provider/marathon/marathon.go

```diff
@@ -420,6 +447,187 @@ func (p *Provider) hasCircuitBreakerLabels(application marathon.Application) boo
 	return ok
 }
 
+// Map of services properties
+// we can get it with label[serviceName][propertyName] and we got the propertyValue
+type labelServiceProperties map[string]map[string]string
+
+// Check if for the given task and applications, we find labels that are defining services
+func (p *Provider) hasTaskServices(task marathon.Task, applications []marathon.Application) bool {
+	application, err := getApplication(task, applications)
+	if err != nil {
+		log.Errorf("Unable to get marathon application from task %s", task.AppID)
+		return false
+	}
+	return len(extractServicesLabels(application.Labels)) > 0
+}
+
+// Check if for the given application, we find labels that are defining services
+func (p *Provider) hasServices(application marathon.Application) bool {
+	return len(extractServicesLabels(application.Labels)) > 0
+}
+
+// Extract the service labels from application labels of dockerData struct
+func extractServicesLabels(labels *map[string]string) labelServiceProperties {
+	v := make(labelServiceProperties)
+
+	for index, serviceProperty := range *labels {
+		matches := servicesPropertiesRegexp.FindStringSubmatch(index)
+		if matches != nil {
+			result := make(map[string]string)
+			for i, name := range servicesPropertiesRegexp.SubexpNames() {
+				if i != 0 {
+					result[name] = matches[i]
+				}
+			}
+			serviceName := result["service_name"]
+			if _, ok := v[serviceName]; !ok {
+				v[serviceName] = make(map[string]string)
+			}
+			v[serviceName][result["property_name"]] = serviceProperty
+		}
+	}
+
+	return v
+}
+
+// Gets the entry for a service label searching in all labels of the given application
+func getApplicationServiceLabel(application marathon.Application, serviceName string, entry string) (string, bool) {
+	value, ok := extractServicesLabels(application.Labels)[serviceName][entry]
```

Comment:
It would not panic (just tested the usecase on play.golang.org)  The `ok` would equal `false` and `value` would have default `string` value of `""`

We do check for `ok` in all the places we call `getApplicationServiceLabel`, like `getServicePriority` for example, so that covers the use-case (IMHO)